### ISSUE TITLE
Add notification Widget

### ISSUE THREAD

**Issue by LeftyDextrous** [davidtakac/bura#18]
Hello,

Recently discovered your app. Very neat, very easy to read and figure out what the day will feel like.

That said, it'd be nice to have a notification that provides the temperature and some other information upon expansion. This is the killer feature I need in a weather app.

Cheers,
Lefty

**Comment by davidtakac** [comment]
Hey Lefty, thank you for the feedback. This is a good idea. A friend of mine suggested the same thing, and I agree. This is planned :)

What other information would you like? Personally, I don't use such features, so it would be great to gather some opinions.

**Comment by LeftyDextrous** [comment]
Well, one useful feature is to have the notification icon be the tempurature, as it allows me to quickly check the weather from the top bar w/o opening my notifications. The last weather app I used gave a forecast for the next few hours but it'd be just as useful to have a brief summary of the _current_ conditions, i.e. temperature, precip info, wind, humidity instead, since most of the time I check the notification only if I'm going outside for a commute or walk or similar unplanned excursion and don't plan to be outside for an event or anything.

**Comment by davidtakac** [comment]
Notification icon being the temperature is a pretty involved task because all of those would have to be separate images. I'm also not sure if temperature formatting is different in some locales, so that might be another can of worms. 

What could be done instead is having the icon be one of the condition icons (rain, sunny, overcast etc.) and then have the basic info in text. 

Could you tell me what application you used had that feature?

**Comment by LeftyDextrous** [comment]
https://github.com/breezy-weather/breezy-weather

Breezy Weather. I have no idea how they did, well, anything, since I know precious little about programming, but it's open-source under LGPL, so you can look into how they did it

**Comment by LeftyDextrous** [comment]
Perhaps the image could be generated on-the-fly from text input but I'd imagine that'd need another library or extra work.

**Comment by davidtakac** [comment]
Yeah as soon as I wrote that it would be involved work, I thought the same thing. Probably still a bit involved 😄  But fun to think about. 

I'll see how Breezy did it, thanks!

**Comment by LeftyDextrous** [comment]
Thank you for developing this app! There's a reason I switched over and it's because it's so clean and easy to read at a glance!

Apr 2, 2024 18:06:30 David Takač ***@***.***>:

**Comment by papjul** [comment]
Hi, happy to see you back with a new weather app, David! I really liked Prognoza, especially the technical part which you're very good at :)

Initially, Geometric Weather used one image for each value from -140° to 140° (which btw didn't work when user chose Kelvin as its unit).

We moved to create on-the-fly the bitmap files in Breezy Weather.

We had to bundle our own font because the default font lacks a bit of readability and can be unpredictable with some custom ROM with different font.

We also had to put a workaround to take care of a leading whitespace.

And here is the relevant part of code of Breezy Weather:
https://github.com/breezy-weather/breezy-weather/blob/v5.1.8/app/src/main/java/org/breezyweather/theme/resource/ResourceHelper.kt#L153-L178

Good luck! :)

**Comment by LeftyDextrous** [comment]
Oh my gosh I love the open source community

**Comment by davidtakac** [comment]
Thank you Julien, you are very kind. I love what you've done with Breezy Weather. And thank you for sharing your on-the-fly temperature icon logic. It is surprisingly simple 🙌
